api-gateway/kong/plugins/pepkong/kong/plugins/pepkong/handler.lua

```diff
@@ -0,0 +1,209 @@
+        })
+    end
+
+    -- Decode token
+    local jwt, err = jwt_decoder:new(token)
+    if err then
+        return false, {
+            status = 401,
+            message = "Bad token; " .. tostring(err)
+        }
+    end
+
+    -- Invoke PDP/Keycloak
+    local params = build_form_params(conf.resource, conf.scopes[kong.request.get_method()])
+    local token_endpoint = jwt.claims.iss .. "/protocol/openid-connect/token"
+
+    local protocol = string.sub(token_endpoint, 0, 5)
+
+    kong.log.debug('Invoke PDP/Keycloak in endpoint: ', token_endpoint)
+    kong.log.debug('Invoke PDP/Keycloak with params: ', params)
+
+    local response = {}
+
+    local header_request = {
+        ["Authorization"] = "Bearer " .. token,
+        ["Content-Type"] = "application/x-www-form-urlencoded",
+        ["Accept"] = 'application/json',
+        ["content-length"] = string.len(params)
+    }
+
+    local source_request = ltn12.source.string(params)
+    local method_request = "POST"
+    local sink_request = ltn12.sink.table(response)
+
+    local base_request = {
+        method = method_request,
+        url = token_endpoint,
+        source = source_request,
+        headers = header_request,
+        sink = sink_request
+    }
+
+    local do_request = nil
+
+    -- This lib "ssl.https" doesn't use certificate verification as we explicitly specified
+    -- base_request['verify']="none" in the parameter list. Certificate verification allows the client
+    -- (this script in this case) to confirm that the site it is connected to has access
+    -- to a valid certificate, signed by a certificate authority.
+    -- This is the same process that your web browser does for you
+    -- when you connect to a website over https. To enable certificate
+    -- verification in SSL verify="none" needs to be replaced with verify="peer" or "client_once".
+    -- This is not enough, however: if you run this code with verify="peer" or "client_once",
+    -- you will get: `certificate verify failed error`.
+    -- This error indicates that we requested certificate verification,
+    -- but have no means to verify this certificate as we don't have any certificate
+    -- authority certificates we can use to validate the signature on
+    -- the certificate provided by the website we are connecting to.
+    -- To do that, we need to find and download the file with these certificates;
+    -- it comes with your browser, but can also be found online.
+    -- For example, this internal cert from kong "/etc/ssl/certs/ca-certificates.crt"
+    -- I was loading an internal kong base_request['cafile']="/etc/ssl/certs/ca-certificates.crt" and verify="peer",
+    -- but that was what I was doing greatly increase the memory
+    -- consumption of the kong container, reaching more than 1 gb without decreasing. We have to think about how to solve this.
+    -- https://github.com/brunoos/luasec/wiki
+    -- https://github.com/brunoos/luasec/wiki/LuaSec-1.0.x
```

Comment:
Despite these changes, it appears that memory consumption increases over time, and does not appear to decrease.

Reply:
But it doesn't even come close to how high consumption was before.